balancer/leastrequest/leastrequest.go

```diff
@@ -155,15 +155,19 @@ type picker struct {
 
 func (p *picker) Pick(balancer.PickInfo) (balancer.PickResult, error) {
 	var pickedSC *scWithRPCCount
+	var pickedSCNumRPCs int32
 	for i := 0; i < int(p.choiceCount); i++ {
 		index := grpcranduint32() % uint32(len(p.subConns))
 		sc := p.subConns[index]
 		if pickedSC == nil {
 			pickedSC = &sc
+			pickedSCNumRPCs = pickedSC.numRPCs.Load()
 			continue
 		}
-		if sc.numRPCs.Load() < pickedSC.numRPCs.Load() {
+		scNumRPCs := sc.numRPCs.Load()
+		if scNumRPCs < pickedSCNumRPCs {
```

Comment:
Optional: these can be merged into a compound if if desired.  Or even simpler:

```go
sc := p.subConns[index]
n := sc.numRPCs.Load()
if pickedSC == nil || n < pickedSCNumRPCs {
	pickedSC = &sc
	pickedSCNumRPCs = n
}
```

Reply:
Ah, beautiful suggestion :D. I tried breaking it but it works and is much cleaner. Thanks.